src/utils/redux/slices/beaconSlice.ts

```diff
@@ -1,21 +1,20 @@
 import { NetworkType } from "@airgap/beacon-wallet";
 import { createSlice } from "@reduxjs/toolkit";
+import { setWith } from "lodash";
 
 import { RawPkh } from "../../../types/Address";
 
-export type DAppConnectionInfo = {
-  accountPkh: RawPkh;
-  networkType: NetworkType;
-};
-
-type State = Record<string, DAppConnectionInfo>;
+type State = Record<string, Record<RawPkh, NetworkType>>;
```

Comment:
this needs a data migration

Reply:
Yep, thanks for the reminder

Reply:
Actually I'm not sure how to create migration here, given that we alter state type & not simply adding / changing data.

@serjonya-trili could you please help me with that?

Should I save the old state, reset State & then fill it with the saved data?

Reply:
e.g. something like this:

```
 3: (state: any) => produce(state, (draft: any) => {
    const old = draft;
    draft = {};
    Object.keys(old).forEach((dAppId: string) => {
      const savedData = old[dAppId];
      setWith(draft, [dAppId, savedData.accountPkh], savedData.networkType);
    });
  }),
```

Reply:
you can find some examples on how to use produce [here](https://immerjs.github.io/immer/produce)